synapse/handlers/sso.py

```diff
@@ -702,8 +708,106 @@ async def _register_mapped_user(
         await self._store.record_user_external_id(
             auth_provider_id, remote_user_id, registered_user_id
         )
+
+        # Set avatar, if available
+        if attributes.picture:
+            await self.set_avatar(registered_user_id, attributes.picture)
+
         return registered_user_id
 
+    async def set_avatar(self, user_id: str, picture_https_url: str) -> None:
+        """Set avatar of the user.
+
+        This downloads the image file from the url provided, store that in
+        the media repository and then sets the avatar on user's profile.
+
+        Currently, it only supports server configurations which runs media repository
+        within the same process.
```

Comment:
We need to get this written down in the `docs` directory somewhere.

We should enforce this by making this function return early if `config.media.can_load_media_repo is False`, see https://github.com/matrix-org/synapse/blob/f1673866ed8a39d49e2caaa6f4255a3f696bc3b4/synapse/config/repository.py#L115-L129

See https://github.com/matrix-org/synapse/pull/13917#discussion_r982215452 for context.

Reply:
Enforced the check to bail early.

But I am not sure where in docs would be the appropriate place to document this. There isn't a general SSO file, only SSO mapping providers file. What will you suggest?

Reply:
Can we add it to the same place as the `picture_claim` documentation maybe?

Reply:
Added!